force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -102,16 +112,18 @@ public with sharing class ClassesObjectsMethods {
      */
     public static Integer countVowels(String str) {
         // Initialize the result Integer
-
-        /*
-        for () {
+        Integer result = 0;
+        
+        for (Integer i = 0; i < str.length(); i++) {
             // get individual characters from the string - hint use substring or split
+            String character = str.substring(i, i +1);
             // check if the character is a vowel
+            if (character.equals('a') || character.equals('e') || character.equals('i') || character.equals('o') || character.equals('u')) {
             // if it is a vowel, increment the result Integer
+            result++;
         }
-        */
-
-        return null; // Replace null with the variable you used to store the result
+    }
+       return result; // Replace null with the variable you used to store the result
```

Comment:
Logic is solid. When checking for membership, it is often much easier and always more performance to use a Set. 5 or conditions can get confusing to read and test.   See my solution for an example.  
```
  public static Integer countVowels(String str) {
        Integer vowelCount = 0;
        Set<String> vowels = new Set<String>{
                'a', 'e', 'i', 'o', 'u'
        };
        List<String> characters = str.split('');
        for (String character : characters) {
            if (vowels.contains(character.toLowerCase())) {
                vowelCount++;
            }
        }
        return vowelCount;
    }
 ```